code/src/components/app-input-with-icon/AppInputWithIcon.test.tsx

```diff
@@ -0,0 +1,46 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import AppInputWithIcon from "@/components/app-input-with-icon/AppInputWithIcon";
+
+describe("AppInputWithIcon ", () => {
+  test("Should render search input field", () => {
+    render(<AppInputWithIcon placeholder="Search..." />);
+    const searchInputField = screen.getByPlaceholderText("Search...");
+    expect(searchInputField).toBeInTheDocument();
+  });
+
+  test("Should render search icon button", () => {
+    render(<AppInputWithIcon placeholder="Search..." />);
+    const searchButton = screen.getByTestId(/SearchIcon/);
+    expect(searchButton).toBeInTheDocument();
+  });
+
+  test("Should render clear icon button when there is a value", () => {
+    render(<AppInputWithIcon value="test" placeholder="Search..." />);
+    const clearButton = screen.getByTestId(/ClearIcon/i);
+    expect(clearButton).toBeInTheDocument();
+  });
+
+  test("Should call onSearch when search button is clicked", () => {
+    const handleSearch = jest.fn();
+    render(
+      <AppInputWithIcon placeholder="Search..." onSearch={handleSearch} />
+    );
+    const searchButton = screen.getByTestId(/SearchIcon/);
+    fireEvent.click(searchButton);
+    expect(handleSearch).toHaveBeenCalledTimes(1);
```

Comment:
Prefer using `toHaveBeenCalled` or `toHaveBeenCalledWith` in your tests. In my view, these are the best options we can use when testing mocked function. I suggest using `toHaveBeenCalled` here

Reply:
Fixed!